internal/worker/worker.go

```diff
@@ -419,3 +457,28 @@ func (w *BaseWorker) Stop() error {
 	w.Session.Expire()
 	return nil
 }
+
+func (w *BaseWorker) updateLastActivity() {
+	w.lastActivityMux.Lock()
```

Comment:
On each event on channels we're taking locks, I believe this would affect the performance adversely.
Can you please add some before/after benchmarks?
cc: @soumya-codes / @AshwinKul28  please add your inputs on this.

Reply:
Hey
Should I write a benchmark for start method with locks and without locks in updateLastActivity function?

Reply:
Yes, ideally we should analyse the perf impact.
cc: @soumya-codes @AshwinKul28 thoughts?

Reply:
Ok. Will add those benchmarks

Reply:
Hey @lucifercr07 
Can you guide me on how can I mock the base worker in benchmark?

Reply:
Hello @lucifercr07 
I have ran the benchmarks with and without locks in the updateActivityMethod.

With lock:
<img width="763" alt="Screenshot 2024-11-14 at 9 10 44 AM" src="https://github.com/user-attachments/assets/fba4d449-452c-44cf-bedf-64925c60a174">

Without lock:
<img width="777" alt="Screenshot 2024-11-14 at 9 11 13 AM" src="https://github.com/user-attachments/assets/809b399c-12c6-4a62-880b-00e610031662">